src/integrationTest/java/com/hivemq/client/restrictions/Mqtt3SendMaximumIT.java

```diff
@@ -0,0 +1,123 @@
+package com.hivemq.client.restrictions;
+
+import com.hivemq.client.mqtt.MqttGlobalPublishFilter;
+import com.hivemq.client.mqtt.datatypes.MqttQos;
+import com.hivemq.client.mqtt.mqtt3.Mqtt3Client;
+import com.hivemq.client.mqtt.mqtt3.message.connect.Mqtt3Connect;
+import com.hivemq.client.mqtt.mqtt3.message.connect.Mqtt3ConnectBuilder;
+import com.hivemq.client.mqtt.mqtt5.Mqtt5BlockingClient;
+import com.hivemq.client.mqtt.mqtt5.Mqtt5Client;
+import com.hivemq.client.mqtt.mqtt5.message.publish.Mqtt5Publish;
+import com.hivemq.extension.sdk.api.ExtensionMain;
+import com.hivemq.extension.sdk.api.client.ClientContext;
+import com.hivemq.extension.sdk.api.client.parameter.InitializerInput;
+import com.hivemq.extension.sdk.api.interceptor.puback.PubackOutboundInterceptor;
+import com.hivemq.extension.sdk.api.interceptor.puback.parameter.PubackOutboundInput;
+import com.hivemq.extension.sdk.api.interceptor.puback.parameter.PubackOutboundOutput;
+import com.hivemq.extension.sdk.api.parameter.ExtensionStartInput;
+import com.hivemq.extension.sdk.api.parameter.ExtensionStartOutput;
+import com.hivemq.extension.sdk.api.parameter.ExtensionStopInput;
+import com.hivemq.extension.sdk.api.parameter.ExtensionStopOutput;
+import com.hivemq.extension.sdk.api.services.Services;
+import com.hivemq.extension.sdk.api.services.intializer.ClientInitializer;
+import com.hivemq.testcontainer.core.HiveMQExtension;
+import com.hivemq.testcontainer.junit5.HiveMQTestContainerExtension;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.time.Duration;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * @author Yannick Weber
+ */
+public class Mqtt3SendMaximumIT {
+
+    public static final int RECEIVE_MAXIMUM = 10;
+    public static final @NotNull HiveMQExtension NO_PUBACK_EXTENSION = HiveMQExtension.builder()
+            .version("1.0.0")
+            .priority(100)
+            .name("No PUBACK Extension")
+            .id("no-puback-extension")
+            .mainClass(NoPubackExtension.class)
+            .build();
+
+    @RegisterExtension
+    public final @NotNull HiveMQTestContainerExtension hivemq =
+            new HiveMQTestContainerExtension().withExtension(NO_PUBACK_EXTENSION);
+
+    @Test
+    void test() throws InterruptedException {
+
+        final Mqtt3Client publisher = Mqtt3Client.builder().serverPort(hivemq.getMqttPort()).build();
+        publisher.toBlocking().connectWith().restrictions().sendMaximum(RECEIVE_MAXIMUM).applyRestrictions().send();
+
+        Mqtt3Connect.builder().restrictions().sendMaximum(10).sendMaximumPacketSize().applyRestrictions().build();
```

Comment:
.sendMaximumPacketSize() whats that for?

Reply:
and why are you not using the return value?

Reply:
This line should be removed, because it was just done for creating an example for PM.